accepted/future-releases/class-modifiers/feature-specification.md

```diff
@@ -525,37 +533,44 @@ mixinModifier     ::= 'sealed' | 'base' | 'interface' | 'final'
 
 It is a compile-time error to:
 
-*   Extend a class marked `interface` or `final` outside of the library where it
-    is declared.
-
-*   Implement a class or mixin marked `base` or `final` outside of the library
+*   Extend a class marked `interface`, `final` or `sealed` outside of the library
     where it is declared.
-
-*   Mix in a mixin marked `interface` or `final` outside of the library where it
-    is declared.
-
+*   Implement the interface of a class or mixin marked `base`, `final` or `sealed`
+    outside of the library where it is declared.
+*   Mix in a mixin or mixin class marked `interface`, `final` or `sealed` outside 
+    of the library where it is declared.
 *   Extend a class marked `base` outside of the library where it is declared
     unless the extending class is marked `base` or `final`. *This ensures that a
     subtype can't escape the `base` restriction of its supertype by offering its
     _own_ interface that could then be implemented without inheriting the
-    concrete implementation from the supertype.*
-
+    concrete implementation from the supertype.* <!-- Needs to account for `sealed` -->
 *   Mix in a mixin or mixin class marked `base` outside of the library where it
     is declared unless the class mixing it in is marked `base` or `final`. *As
     with the previous rule, ensures you can't get a backdoor interface on a
     mixin that doesn't want to expose one.*
-
 *   Mix in a class not marked `mixin` outside of the library where it is
     declared, unless the class declaration being used as a mixin is in a library
     whose language version is older than the version this feature ships in.
-
+*   Mix in a class not marked `mixin` inside the library where it is declared,
+    if the class has a superclass other than `Object` or the class declares a 
+    non-trivial generative constructor. If the library has a language version
+    older than the version this feature ships int, the class must not declare
+    any generative constructor.
 *   Apply `mixin` to a class whose superclass is not `Object` or that declares a
-    generative constructor. *A `mixin class` states that you intend the class to
-    be mixed in, which is inconsistent with defining a class that can't be used
+    _non-trivial generative constructor_. 
+    *A `mixin class` states that you intend the class to
+    be mixed in, which is inconsistent with defining a class that cannot be used
     as a mixin. Note that this means that `mixin` on a class becomes a helpful
     reminder to ensure that you don't inadvertently break your class's ability
     to be used as a mixin.*
```

Comment:
I'm finding it pretty hard to parse the increasingly complex set of restrictions around when you can apply a mixin. Maybe it would be easier to specify when it's *allowed* and say that it's a compile-time error otherwise.

Reply:
Applying a *mixin* is easy. It's using a non-mixin class as a mixin that's complicated (so why do we even allow it in new code? I'm not sold on allowing you to mix in plain classes from your own library, because I don't see `mixin` as a modifier on the `class` that other classes lack. I see `mixin class` as two declarations in one, and other `class` declarations are just not defining mixins, so mixing them in makes as much sense as extending mixins.)

Anyway, you can mix in a (mixin derived from a) class when:
* The class is in a library with a language version before this feature, and
    * The class has `Object` as superclass and declares no (generative?) constructors.
* The class is in the same library, which has a language version including this feature, and
    * The class has `Object` as superclass and declares no non-trivial generative constructors.
    
A `mixin class` must satisfy the latter requirements.

I think the items I'm converging on say the same thing as negatives, in about the same verbosity.